src/components/ProductStorePopover.vue

```diff
@@ -76,7 +76,7 @@ export default defineComponent({
               primaryFacilityGroupId: ''
             })
           }
-
+          await this.store.dispatch('facility/updateCurrentFacility', { ...this.current, primaryFacilityGroupId: '' })
```

Comment:
We should update the current facility to primaryFacilityGroupId='' only when updateFacility api is successful. So move the code inside the if block.